src/main/java/io/asyncer/r2dbc/mysql/QueryFlow.java

```diff
@@ -1106,25 +1112,37 @@ final class StartTransactionState extends AbstractTransactionState {
 
     private static final int START_TRANSACTION = 4;
 
-    private final long lockWaitTimeout;
-
-    @Nullable
-    private final IsolationLevel isolationLevel;
+    private static final int CANCEL = 8;
```

Comment:
I'm not sure if the `CANCEL` flag is necessary. Currently, if `cancelTasks` returns `true` it means "cancel the current request/exchange", then `CANCEL` flag will not be actually used.

So I think we can just remove the flag, and use `0` for the bitmap value if it should be cancelled.

Or if it should be used, we can change `cancelTasks` to return `void`, then make `TransactionBatchExchangeable#subscribe` and `TransactionMultiExchangeable#subscribe` check the calculated flags, if `this.tasks` is `CANCEL` then send ping message.